### PR TITLE
[abseil] Bump to 20230802.0

### LINKED ISSUE #33097: [abseil] update to LTS 20230802
### Library name

abseil

### Library description

Abseil LTS 20230802

### Source repository URL

https://github.com/abseil/abseil-cpp.git

### Project homepage (if different from the source repository)

_No response_

### Anything else that is useful to know when adding (such as optional features the library may have that should be included)

_No response_

### PR DESCRIPTION
**Pull request by xiaozhuai**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

Close #33097

### CODE REVIEW

File: ports/abseil/portfile.cmake [hunk 1]
```diff
@@ -44,16 +41,20 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share"
...
     )
+    vcpkg_replace_string("${file}"
+        "defined(ABSL_CONSUME_DLL)" "1"
+    )
+endfunction(fix_dll_definitions file)
+
+if (VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
+    file(GLOB_RECURSE headers "${CURRENT_PACKAGES_DIR}/include/absl/*.h")
+    foreach(header ${headers})
+        fix_dll_definitions("${header}")
```
Comment (dg0yt) [inline ports/abseil/portfile.cmake:56]: Can't this be a straight
```suggestion
        vcpkg_replace_string("${header}" "defined(ABSL_CONSUME_DLL)" "(1)")
```

File: ports/abseil/portfile.cmake [hunk 2]
```diff
@@ -44,16 +41,20 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share"
...
+function(fix_dll_definitions file)
+    vcpkg_replace_string("${file}"
         "!defined(ABSL_CONSUME_DLL)" "0"
     )
+    vcpkg_replace_string("${file}"
+        "defined(ABSL_CONSUME_DLL)" "1"
+    )
+endfunction(fix_dll_definitions file)
+
+if (VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
```
Comment (dg0yt) [inline ports/abseil/portfile.cmake:53]:
```suggestion
if (VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
```

File: ports/abseil/portfile.cmake [hunk 3]
```diff
@@ -44,16 +41,20 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share"
...
+    file(GLOB_RECURSE headers "${CURRENT_PACKAGES_DIR}/include/absl/*.h")
+    foreach(header ${headers})
```
Comment (dg0yt) [inline ports/abseil/portfile.cmake:55]:
```suggestion
    foreach(header IN LISTS headers)

```

**Comment by MonicaLiu0311** [comment]
All features are tested successfully in the following triplet:
```
x86-windows
x64-windows
x64-windows-static
```

The usage test passed on `x64-windows` (header files found):
```
abseil provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(absl CONFIG REQUIRED)
    # note: 174 additional targets are not displayed.
    target_link_libraries(main PRIVATE absl::any absl::log absl::base absl::bits)
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**